skyportal/handlers/api/internal/token.py

```diff
@@ -35,8 +35,7 @@ def post(self):
         data = self.get_json()
 
         user = User.query.filter(User.username == self.current_user.username).first()
-        user_acls = {acl.id for acl in user.acls}
-        token_acls = set(data['acls']) & user_acls
+        token_acls = set(data['acls']) & set(user.permissions)
```

Comment:
I think this should throw an exception when the user posts an ACL they dont have rather than silently not adding it to the token